tests/unit/io/test_eventletreactor.py

```diff
@@ -18,50 +18,32 @@
 except ImportError:
     import unittest # noqa
 
-from tests.unit.io.utils import submit_and_wait_for_completion, TimerCallback
-from tests import is_eventlet_monkey_patched
+from tests.unit.io.utils import TimerConnectionTests
+from tests.unit.io.eventlet_utils import restore_saved_module
+from tests import notpypy
+from tests import notmonkeypatch
+
 import time
+from eventlet import monkey_patch, kill
 
 try:
     from cassandra.io.eventletreactor import EventletConnection
 except ImportError:
     EventletConnection = None  # noqa
 
-
-class EventletTimerTest(unittest.TestCase):
+@unittest.skipUnless(EventletConnection is not None, "Skpping the eventlet tests because it's not installed")
+@notmonkeypatch
+# There are some issues with some versions of pypy and eventlet
+@notpypy
+class EventletTimerTest(unittest.TestCase, TimerConnectionTests):
 
     def setUp(self):
-        if EventletConnection is None:
-            raise unittest.SkipTest("Eventlet libraries not available")
-        if not is_eventlet_monkey_patched():
-            raise unittest.SkipTest("Can't test eventlet without monkey patching")
+        self.connection_class = EventletConnection
+        # We only to patch the time module
+        monkey_patch(time=True)
```

Comment:
Sure, I've pushed a slightly different version using setUpClass, also I've removed the tearDown as I'd rather it is explicit that the unpatch hasn't happened. Please let me know if you see any issues with this.